### PR TITLE
replaces redcarpet with kramdown, to avoid jruby foo

### PR DESCRIPTION
**Pull request by LeFnord**
relates to: https://github.com/danger/danger/issues/394

### CODE REVIEW

**Comment by LeFnord** [comment]
… its so cool 😄

File: lib/danger/helpers/comments_helper.rb [hunk 1]
```diff
@@ -1,10 +1,11 @@
-require "redcarpet"
+# require "redcarpet"
+require "kramdown"
```
Comment (orta) [inline lib/danger/helpers/comments_helper.rb]: Cool, yeah, this looks good - can you remove this moment then we're good to go 👍

File: lib/danger/helpers/comments_helper.rb [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
 
 module Danger
   module Helpers
     module CommentsHelper
-      def markdown_parser
-        @markdown_parser ||= Redcarpet::Markdown.new(Redcarpet::Render::HTML.new(hard_wrap: true), no_intra_emphasis: true)
+      def markdown_parser(text)
+        Kramdown::Document.new(text, input: "GFM")
```
Comment (DanToml) [inline lib/danger/helpers/comments_helper.rb]: We probably still want to cache this?

Comment (orta) [inline lib/danger/helpers/comments_helper.rb]: Yeah - good call, can't hurt 👍

**Comment by orta** [comment]
( and a rebase - sorry about that 👯 )

File: lib/danger/helpers/comments_helper.rb (hunk 2 above)
Comment (LeFnord) [inline lib/danger/helpers/comments_helper.rb]: normally yeap, but Kramdown behaves a bit different to Redcarpet, it needs a string for initialising, so by caching the object it has always the same string to render, think this is not what would be expected

**Comment by orta** [comment]
oh damn, err, sorry, I should have merged this before I merged #399 :/
can I get another rebase

**Comment by orta** [comment]
Thanks @LeFnord  - we have [policy of offering](https://github.com/Moya/contributors) org access after a PR, would you like in? No pressure.

**Comment by LeFnord** [comment]
thanks for the offer @orta, but I think I can also contribute without to be a member 😉 
and yeah maybe think about these policy and differ between small changes – such as this one, and more significant one … similar to the grape membership

**Comment by orta** [comment]
Thanks for the recommendation, but we've thought a long time about the policies for running larger OSS projects and are pretty happy with what we've come up with 👍

**Pull request merged**